NitroxServer/GameLogic/Bases/BaseManager.cs

```diff
@@ -9,153 +9,169 @@ namespace NitroxServer.GameLogic.Bases
+                    NitroxModel.Logger.Log.Error("BasePieceConstructionCompleted - Received ConstructionCompleted for unknown NitroxID: " + id);
                 }
             }
         }
 
         public void BasePieceDeconstructionBegin(NitroxId id)
         {
+
+#if TRACE && BUILDING
+            NitroxModel.Logger.Log.Debug("BasePieceDeconstructionBegin - id: " + id + " - baseId: " + baseId);
+#endif
+
             BasePiece basePiece;
 
-            lock (completedBasePieceHistory)
+            lock (allBasePieces)
             {
-                basePiece = completedBasePieceHistory.Find(piece => piece.Id == id);
+                basePiece = allBasePieces.Find(piece => piece.Id == id);
 
                 if (basePiece != null)
                 {
                     basePiece.ConstructionAmount = 0.95f;
                     basePiece.ConstructionCompleted = false;
-                    completedBasePieceHistory.Remove(basePiece);
-
-                    lock(partiallyConstructedPiecesById)
-                    {
-                        partiallyConstructedPiecesById[basePiece.Id] = basePiece;
-                    }
                 }
-            }        
+                else
+                {
+                    NitroxModel.Logger.Log.Error("BasePieceDeconstructionBegin - Received DeconstructionBegin for unknown NitroxID: " + id);
+                }
+            }
         }
 
         public void BasePieceDeconstructionCompleted(NitroxId id)
         {
-            lock (partiallyConstructedPiecesById)
+
+#if TRACE && BUILDING
+            NitroxModel.Logger.Log.Debug("BasePieceDeconstructionCompleted - id: " + id);
+#endif
+
+            BasePiece basePiece;
+
+            lock (allBasePieces)
             {
-                partiallyConstructedPiecesById.Remove(id);
+                basePiece = allBasePieces.Find(piece => piece.Id == id);
+
+                if (basePiece != null)
+                {
+                    allBasePieces.Remove(basePiece);
+                }
+                else
+                {
+                    NitroxModel.Logger.Log.Error("BasePieceDeconstructionCompleted - Received DeconstructionCompleted for unknown NitroxID: " + id);
+                }
             }
         }
 
         public void UpdateBasePieceMetadata(NitroxId id, BasePieceMetadata metadata)
         {
             BasePiece basePiece;
 
-            lock (completedBasePieceHistory)
+            lock (allBasePieces)
             {
-                basePiece = completedBasePieceHistory.Find(piece => piece.Id == id);
+                basePiece = allBasePieces.Find(piece => piece.Id == id);
 
                 if (basePiece != null)
                 {
                     basePiece.Metadata = Optional.OfNullable(metadata);
                 }
+                else
+                {
+                    NitroxModel.Logger.Log.Error("UpdateBasePieceMetadata - Received UpdateBasePieceMetadata for unknown NitroxID: " + id);
```

Comment:
minor: no need to fully qualify these (here and everywhere)

Reply:
In some classes a normal Log output is not needed only for my trace outputs. If i make a using in these classes, this will lead to showing the using as not needed when the trace define is not set, so i prefer to use the full qualifier in the trace blocks. 

In this case here, you are right, this can be solved via a using.